src/index.tsx

```diff
@@ -179,3 +179,46 @@ export const useLazyRef = <T extends any>(
 
   return ref as MutableRefObject<T>;
 };
+
+/**
+ * Starts loading a worker immediately and handle loading, error and result states.
+ *
+ * See [[useWorker]] for more details
+ *
+ * @param worker An asynchronous function that returns data, which is saved into a state
+ * @param initialValue The data's initial value. Optional when Data may be undefined
+ */
+export const useWorkerLoad = <Data extends any>(
+  worker: () => Promise<Data>,
```

Comment:
What happens if the worker is a function that depends on other data? Shouldn't we have a `dependencies?: Dependencies` here?

Reply:
If the idea is to not allow the `worker` to be updated, shouldn't it be explained some how in the JSDoc?

Reply:
The idea is that the `worker` function is only executed when the component gets mounted, so there's no need to update its dependencies. That's also why the callback is not exposed.

I added the `retry` thinking of a scenario with errors, but I don't know if it should exist when `error` is `undefined`. What do you think? The way that it is now, maybe `reload` or `rework` would be better names for it. If we decide to allow the user to call `retry` whenever they want, then we may need to add `worker` as a dependency of `useWorker`, for the case where the function is updated with new values.

Reply:
If the idea is to retry only on error, maybe creating a boolean `retryOnError` something like that and handling inside `useWorkerLoad` would avoid requiring dependencies.

Reply:
But when would `retry` happen in that case? Immediately? I don't know if that's the best case. How would we prevent infinite retries? IMO, it'd be better if the user triggered a new retry

Reply:
Idea: what if `error` was typed in the return as: `error?: { value: Error; retry: () => void }`. With the coalescing syntax, it'd be simple to access `error?.value` and `error?.retry()`

Reply:
🤔 I like this idea